### PR TITLE
fix out-of-bound array access, some minor clean up

### PR DESCRIPTION
**Pull request by rabauke**

### CODE REVIEW

File: src/win/rlimit.c [hunk 1]
```diff
@@ -87,17 +82,15 @@ int getrlimit(int resource, struct rlimit *rlp)
...
-  {
-    iRet = EINVAL;
-  }
+  if(resource < 0 || resource >= RLIM_NLIMITS)
+    return EINVAL;
 
   //
   // Return both rlim_cur and rlim_max
   //
   *rlp = rlimits[resource];
```
Comment (rabauke) [inline src/win/rlimit.c:91]: Code checks if `resource` is a valid index, but accesses the array `rlimits` not matter if `resource` not out-of-bound or not.

File: src/win/rlimit.c [hunk 2]
```diff
@@ -129,99 +118,79 @@ int setrlimit(int resource, const struct rlimit *rlp)
...
     rlimits[resource].rlim_cur = rlp->rlim_cur;
   else
-    iRet = EINVAL;
+    return EINVAL;
   //
   // We should not let the user set the max value.  However,
   // since currently there is no defined source for initial
   // values, we will let the user change the max value.
   //
   rlimits[resource].rlim_max = rlp->rlim_max;
```
Comment (rabauke) [inline src/win/rlimit.c:127]: Same as above.

File: src/win/rlimit.c [hunk 3]
```diff
@@ -129,99 +118,79 @@ int setrlimit(int resource, const struct rlimit *rlp)
...
-  }
-  else
-  {
-    //
-    // Do the actual write the user requested
-    //
-    dwWritten = _write(handle, buffer, count);
-  }
-  return dwWritten;
+    return = 0;
```
Comment (johnny-bit) [inline src/win/rlimit.c:188]: `return = 0` happens ;) this is the one CI complains about ;)

File: src/win/rlimit.c [hunk 4]
```diff
@@ -129,99 +118,79 @@ int setrlimit(int resource, const struct rlimit *rlp)
...
-  liPosition = (__int64)position;
+  __int64 liPosition = (__int64)ftell(stream);
 
   //
   // Check to make sure the write will not exceed the RLIMIT_FSIZE limit.
   //
-  if((liPosition + liByteCount) > rlimits[RLIMIT_FSIZE].rlim_cur)
-  {
+  if(liPosition + liByteCount > rlimits[RLIMIT_FSIZE].rlim_cur)
     //
```
Comment (TurboGit) [inline src/win/rlimit.c:153]: If I read correctly the diff I would keep the comments + return 0 inside { } for clarity.

Comment (rabauke) [reply src/win/rlimit.c:153]: Actually, I would incline to remove most of of the comments completely. Comments are useful to provide broader context information that is not easily to infer from the code itself. Code comments should not explain the code's obvious functionality as it is the case here. Code should be self-explanatory as far as possible.

Nevertheless, I will just add the braces again.

File: src/win/rlimit.c (hunk 3 above)
Comment (rabauke) [reply src/win/rlimit.c]: This code is Windows specific, but I am on Linux. That's why I was not able to compile it for myself. Luckily, we have an automated build chain for all architectures.

**Review by TurboGit — APPROVED** [review]
Looks good to me now, thanks!

**Pull request merged**